h/views/api_auth.py

```diff
@@ -2,102 +2,88 @@
-
-        Depending on the "response_mode" parameter the auth code will be
-        delivered either via a redirect or via a `postMessage` call to the
-        opening window.
-        """
-        authclient = self._check_params()
-
+        # We don't support scopes at the moment, but oauthlib does need a scope,
+        # so we're adding a hard-coded `annotation:read` and `annotation:write` scopes.
+        scopes = ['annotation:read', 'annotation:write']
         user = self.user_svc.fetch(self.request.authenticated_userid)
+        credentials = {'user': user}
 
-        # Create an "authorization code" for the response.
-        # This is in fact just a JWT grant token since auth codes have not yet
-        # been implemented.
-        auth_code = self.oauth_svc.create_grant_token(user, authclient)
-
-        params = self.request.params
-
-        return {'code': auth_code,
-                # Once authclients have an Origin property, that should be used
-                # instead here.
-                'origin': self.request.host_url,
-                'state': params.get('state')}
+        headers, body, status = self.oauth.create_authorization_response(
+                self.request.url, scopes=scopes, credentials=credentials)
 
-    def _check_params(self):
-        """
-        Check parameters for the authorization request.
+        if 'Location' in headers:
+            return exception_response(status, location=headers['Location'])
+        else:
+            log.error('Created OAuth 2 authorization code, but got no redirect location')
+            raise HTTPInternalServerError()
 
-        If the parameters are valid, returns an authclient.
-        Otherwise, raises an exception.
-        """
-        params = self.request.params
 
-        client_id = params.get('client_id', '')
-        authclient = self.oauth_svc.get_authclient_by_id(client_id)
-        if not authclient:
-            raise HTTPBadRequest('Unknown client ID "{}"'.format(client_id))
-
-        if authclient.authority != self.request.authority:
-            raise HTTPBadRequest('Client "{}" not allowed to authorize "{}" users'.format(
-                                 client_id, self.request.authority))
-
-        response_type = params.get('response_type')
-        if response_type != 'code':
-            raise HTTPBadRequest('Unsupported response type "{}"'
-                                 .format(response_type))
-
-        response_mode = params.get('response_mode', 'query')
-        if response_mode != 'web_message':
-            raise HTTPBadRequest('Unsupported response mode "{}"'.format(response_mode))
+@view_defaults(renderer='h:templates/oauth/error.html.jinja2')
+class OAuthAuthorizeErrorController(object):
+    def __init__(self, context, request):
+        self.context = context
+        self.request = request
 
-        return authclient
+    @view_config(context=InvalidRequestFatalError)
+    @view_config(context=InvalidRequestError)
```

Comment:
I couldn't find anything. According to the standard, it is required and the request must fail when it's missing (see: https://hyp.is/74MpXmsLEee4vhcFDgNKQA/tools.ietf.org/html/rfc6749).

My best guess is that the oauthlib maintainers missed this line when they reviewed the PR which introduced this (https://github.com/idan/oauthlib/pull/358).